src/components/NcCheckboxRadioSwitch/NcCheckboxContent.vue

```diff
@@ -39,9 +40,20 @@
 			</slot>
 		</span>
 
-		<span v-if="$slots.default" class="checkbox-content__text" :class="[textClass]">
-			<!-- @slot The checkbox/radio label -->
-			<slot />
+		<span class="checkbox-content__wrapper">
+			<span
+				v-if="$slots.default"
+				:id="labelId"
+				class="checkbox-content__text"
+				:class="textClass">
+				<!-- @slot The checkbox/radio label -->
+				<slot />
+			</span>
+			<em v-if="!isButtonType && ($slots.description || description)" :id="descriptionId" class="checkbox-content__description">
+				<slot name="description">
+					{{ description }}
+				</slot>
```

Comment:
I wonder if we should simplify this like

```suggestion
				<slot name="description" />
```

And also completely remove the description prop from this component

Reply:
Why does the sub component also need to provide both? Only having the slot simplifies the sub-component by only care about slot not slot or prop.